### PR TITLE
Add biFlatMap to EitherT

### LINKED ISSUE #2438: Add biFlatMap to EitherT
`EitherT` has a `bimap` but no `biFlatMap`. This would be useful when chaining computations and needing to process both sides of the `Either` with functions that return an `EitherT`. For example:

```scala
def findVertex(id: String): EitherT[Future, APIError, Vertex]
def addVertex(...): EitherT[Future, APIError, String]
def doSomethingWithVertex(...): EitherT[Future, APIError, String]

findVertex("id").biFlatMap( {
  case _: APIError.NotFound => addNewVertex(...)
  case error => EitherT.left(error)
},
vertex => doSomethingWithVertex(vertex)
)
```

### PR DESCRIPTION
**Pull request by agajek**
Fixes #2438

### CODE REVIEW

File: core/src/main/scala/cats/data/EitherT.scala [hunk 1]
```diff
@@ -77,6 +77,13 @@ final case class EitherT[F[_], A, B](value: F[Either[A, B]]) {
                                                          applicativeG: Applicative[G]): G[EitherT[F, C, D]] =
     applicativeG.map(traverseF.traverse(value)(axb => Bitraverse[Either].bitraverse(axb)(f, g)))(EitherT.apply)
 
+  def biFlatMap[AA >: A, BB >: B](fa: A => EitherT[F, AA, BB],
+                                  fb: B => EitherT[F, AA, BB])(implicit F: Monad[F]): EitherT[F, AA, BB] =
```
Comment (ceedubs) [inline core/src/main/scala/cats/data/EitherT.scala]:
```suggestion
                                  fb: B => EitherT[F, AA, BB])(implicit F: FlatMap[F]): EitherT[F, AA, BB] =
```

**Comment by agajek** [comment]
@ceedubs is it possible to rerun travis' build? It indicates some memory problem of host rather than issues with my code.

**Comment by ceedubs** [comment]
We have a 2-approver rule for merging non-documentation PRs. @kailuowang or @LukaJCB would one of you want to take a look?

**Comment by ochrons** [comment]
A question on naming, should this actually be `biflatMap` instead of `biFlatMap` as there are `bimap` and `bitraverse` with no capitalization in the middle?

File: core/src/main/scala/cats/data/EitherT.scala [hunk 2]
```diff
@@ -77,6 +77,13 @@ final case class EitherT[F[_], A, B](value: F[Either[A, B]]) {
                                                          applicativeG: Applicative[G]): G[EitherT[F, C, D]] =
     applicativeG.map(traverseF.traverse(value)(axb => Bitraverse[Either].bitraverse(axb)(f, g)))(EitherT.apply)
 
+  def biFlatMap[AA >: A, BB >: B](fa: A => EitherT[F, AA, BB],
+                                  fb: B => EitherT[F, AA, BB])(implicit F: FlatMap[F]): EitherT[F, AA, BB] =
+    EitherT(F.flatMap(value) {
+      case Left(a)  => fa(a).value
+      case Right(a) => fb(a).value
```
Comment (phderome) [inline core/src/main/scala/cats/data/EitherT.scala]: would not it be more intuitive to say `case Right(b) => fb(b).value` at L.84 to match with the type `B` fb expects?

**Pull request merged**